scopes/lanes/diff/lane-diff-generator.ts

```diff
@@ -104,4 +133,38 @@ export class LaneDiffGenerator {
     const toLaneName = values.length === 2 ? values[1] : values[0];
     return { fromLaneName, toLaneName };
   }
+
+  private async getDefaultLaneData(ids: BitId[]): Promise<LaneData> {
+    const laneData: LaneData = {
+      name: DEFAULT_LANE,
+      remote: null,
+      components: [],
+      isMerged: null,
+    };
+    for await (const id of ids) {
+      const modelComponent = await this.scope.legacyScope.getModelComponent(id);
+      const laneComponent = {
+        id,
+        head: modelComponent.head as Ref,
+        version: modelComponent.latestVersion(), // should this be latestVersion() or bitId.version.toString()
+      };
+      laneData.components.push(laneComponent);
+    }
+    return laneData;
+  }
+
+  private async mapToLaneData(lane: Lane): Promise<LaneData> {
+    const { name, components, remoteLaneId } = lane;
+    const isMerged = await lane.isFullyMerged(this.scope.legacyScope);
+    return {
+      name,
+      isMerged,
+      components: components.map((lc) => ({
+        id: lc.id,
+        head: lc.head,
+        version: lc.id.version?.toString(),
+      })),
+      remote: (!!remoteLaneId && remoteLaneId.toString()) || null,
```

Comment:
Here, also, I guess no need for the `!!`. Maybe you can just go with `remoteLaneId?.toString()` instead of `(!!remoteLaneId && remoteLaneId.toString())`

Reply:
optional chaining will result the type to be `string | undefined` which won't satisfy the prop type `remote: string | null`.  `!!` coercion can be removed but I will still need to have either `?? null` or `|| null`

Reply:
Correct. I was referring to the first part only `(!!remoteLaneId && remoteLaneId.toString())`. The rest `|| null` is indeed needed for the types. 
Looks good now.